### PR TITLE
Add `apple simulators reset-simulator` command

### LINKED ISSUE #777: Create `apple reset-simulator` command
### Motivation
Currently, we have the `--reset-simulator` switch which wipes the simulator clean.
For some cases (simulator is broken / its memory is full), we might want to call the reset only but it is not possible - always a part of some other command.

### Goal
Add a new command that would reset given simulator.

### PR DESCRIPTION
**Pull request by premun**
Resolves #777

### CODE REVIEW

File: src/Microsoft.DotNet.XHarness.Apple/Orchestration/SimulatorResetOrchestrator.cs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        if (!target.Platform.IsSimulator() || target.Platform.ToRunMode() == RunMode.MacOS)
+        {
+            _consoleLogger.LogError($"The simulator reset action requires a simulator target while {target.AsString()} specified");
+            return Task.FromResult(ExitCode.INVALID_ARGUMENTS);
+        }
+
+        static Task<ExitCode> executeMacCatalystApp(AppBundleInformation appBundleInfo)
+            => throw new InvalidOperationException("reset-simulator command not available on maccatalyst");
+
+        static Task<ExitCode> executeApp(AppBundleInformation appBundleInfo, IDevice device, IDevice? companionDevice)
```
Comment (lpatalas) [inline src/Microsoft.DotNet.XHarness.Apple/Orchestration/SimulatorResetOrchestrator.cs:64]: Both execute methods start with lowercase. Is this intended?

File: tests/Microsoft.DotNet.XHarness.Apple.Tests/Orchestration/SimulatorResetOrchestratorTests.cs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        // Verify
+        Assert.Equal(ExitCode.INVALID_ARGUMENTS, result);
+
+        _deviceFinder.VerifyNoOtherCalls();
+        _appInstaller.VerifyNoOtherCalls();
+        _appUninstaller.VerifyNoOtherCalls();
+    }
+
+    [Fact]
+    public async Task TryMaccatalystResetTest()
```
Comment (lpatalas) [inline tests/Microsoft.DotNet.XHarness.Apple.Tests/Orchestration/SimulatorResetOrchestratorTests.cs:87]:
```suggestion
    public async Task TryMacCatalystResetTest()
```

**Review by lpatalas — APPROVED** [review]
I don't have enough context to review it in depth but what I see looks good :)

File: src/Microsoft.DotNet.XHarness.Apple/Orchestration/SimulatorResetOrchestrator.cs (hunk 1 above)
Comment (premun) [reply src/Microsoft.DotNet.XHarness.Apple/Orchestration/SimulatorResetOrchestrator.cs:64]: They used to be lambdas and I think an auto-refactoring changed them to local methods. Do you know what's the naming convention for these?

Comment (lpatalas) [reply src/Microsoft.DotNet.XHarness.Apple/Orchestration/SimulatorResetOrchestrator.cs:64]: I have no idea. I didn't even notice that these are local functions before you wrote it. I would guess they should be uppercase.

**Pull request merged**